bitvm/src/signatures/utils.rs

```diff
@@ -11,64 +11,95 @@ pub(super) const fn log_base_ceil(n: u32, base: u32) -> u32 {
     res
 }
 
-/// Converts the number to given base, smallest digit being at the start (returns `digit_count` smallest digits of it, and standard base representation of it if `digit_count` = -1)
-pub(super) fn to_digits(mut number: u32, base: u32, digit_count: i32) -> Vec<u32> {
-    let mut digits = Vec::new();
-    if digit_count == -1 {
-        while number > 0 {
-            let digit = number % base;
-            number = (number - digit) / base;
-            digits.push(digit);
-        }
-    } else {
-        digits.reserve(digit_count as usize);
-        for _ in 0..digit_count {
-            let digit = number % base;
-            number = (number - digit) / base;
-            digits.push(digit);
-        }
+/// Converts the given `checksum` into a vector of digits.
+///
+/// There are `n_digits` many digits.
+/// Each digit is a `u32` value in the range `0..base`.
+/// Digits are in big endian order.
+///
+/// ## Panics
```

Comment:
Not in release mode.

Reply:
I changed an `assert!` into a `debug_assert!`. Thanks for the headsup.